localstack-core/localstack/services/lambda_/event_source_mapping/esm_event_processor.py

```diff
@@ -88,12 +94,10 @@ def process_target_stage(self, events: list[dict]) -> None:
                     logLevel=LogLevel.TRACE,
                 )
                 # TODO: handle and log target invocation + stage skipped (when no records present)
-                payload = self.sender.send_events(events)
-                if payload:
+                if response_payload := self.sender.send_events(events):
```

Comment:
It passes on my local but I've triggered the community tests against specifically the feature branch in pro to ensure all works fine: https://github.com/localstack/localstack/actions/runs/10741749751

Reply:
I meant it's not ideal that the Pipes implementation now differs from ESM v2 here because having both test suites was valuable during the migration and keeping implementations in sync helps for consistency (but it's probably nitpicking)